cmd/charm/charmcmd/revoke.go

```diff
@@ -112,7 +112,7 @@ func (c *revokeCommand) Run(ctxt *cmd.Context) error {
 	}
 	defer client.jar.Save()
 
-	if c.id.Revision == -1 {
+	if c.channel != "" {
```

Comment:
Technically this changes behaviour so there should be a test that needs to change too, but I'll let you off this time :)